### ISSUE TITLE
Can we replace conv2d by gconv2d directly?

### ISSUE THREAD

**Issue by a14en9** [tscohen/gconv_experiments#3]
Dear Dr. Cohen.

Can we simply use gconv2d as an replacement of conv2d in any standard architecture, such as VGG16?

Or we have to train the model from scratch?

Best wishes

**Comment by tscohen** [comment]
You would need to train the model from scratch, because a G-Conv has a different number of parameters from a conv. To turn any CNN architecture into a G-CNN, you need to replace every layer with an equivariant counterpart. For VGG, I think it would just amount to replacing each conv layer by a G-conv layer and then re-training. But I always recommend checking whether the architecture you made is actually equivariant, e.g. by feeding an input image and a 90-degree rotated one, and seeing whether the output of the last layer transforms in the way you expect (e.g. by rotating and channel-shuffling).

**Comment by cvhabitat** [comment]
Dear Dr. Cohen.,

Thank you for the information. I don't think that I understand the idea behind of G-Conv. Could you tell me more about it, please. 
From my understanding, G-Conv is porposed to learn transformation equivariant by leveraging the symmetry properties of transfromations. If so, why it cannot garantee to be equivariant with simply replacing a specific layer of standard architecture by a G-Conv? If this not work, how should I understanding the following statement in paper?

"That is, transforming an input x by a transformation g
(forming Tg x) and then passing it through the learned map
Φ should give the same result as first mapping x through Φ
and then transforming the representation."

In other words, all of transformations are achieved by using the permutations of kernels since transformation feature map is more complex. If it is not be equivariant anymore, do you think it can achieve transformation-invariant by using maximum pooling right after G-Conv?

Best wishes

**Comment by a14en9** [comment]
Hi Dr.Cohen,

Thank you for your explanations. I notice that you listed results on Cifar 10 datasets. Have you pretrained your own model on large-scale datasets, such as ImageNet. Is that possible to share your pretrained model and weights on Cloud, please?

Best wishes
